### PR TITLE
Test realm pluginOptions in multiple routes using route and reply interface

### PR DESCRIPTION
**Pull request by geek**
@hueniverse there is a question on if `request.server.realm.pluginOptions` should contain the plugin options for the related realm?  At the moment, this value is an empty object.

I documented another way to get this value in a route, which is using reply.realm.

### CODE REVIEW

File: test/plugin.js [hunk 1]
```diff
@@ -550,6 +550,39 @@ describe('Plugin', () => {
...
+            server.connection({ labels: ['a', 'b'] });
+
+            const test = function (srv, options, next) {
+
+                expect(options.something).to.be.true();
+                expect(srv.realm.pluginOptions).to.equal(options);
+
+                srv.route({ method: 'GET', path: '/', handler: (request, reply) => {
+
+                    expect(request.server.realm.pluginOptions).to.deep.equal(options);
```
Comment (geek) [inline test/plugin.js]: `request.server.realm.pluginOptions` is `{}` here

**Comment by devinivy** [comment]
I would expect `request.server.realm` to be the realm in which the route was registered.  So at a glance this does seem to be a valid issue to me.

On the other hand, is `request.server` the same as `srv.root`?

**Comment by devinivy** [comment]
Nice.  You surfaced some new pieces of the public API to the docs– might as well be a feature!  Do we think the docs listed for `request.server` are specific enough?  It seems to be that it will always be the root server.

**Comment by geek** [comment]
@devinivy I think the current `request.server` are enough.

**Pull request merged**